File: Hayaa.ServiceFrame/ServiceFrameClient/Hayaa.ServicePlatform.Client/Config/AppComponent.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.ServicePlatform.Client.Config
{
    [Serializable]
    internal class AppComponent
    {
        //public int ComponentInstanceId
        //{
        //    get;
        //    set;
        //}
        ///// <summary>
        ///// 程序用户ID灰度发布需要的模式TODO
        ///// </summary>	
        //public List<int> AppUserId
        //{
        //    get;
        //    set;
        //}
        /// <summary>
        /// 程序用户ID
        /// </summary>	
        public int AppUserId
        {
            get;
            set;
        }
        /// <summary>
        /// 组件ID
        /// </summary>		
        public int ComponetId
        {
            get;
            set;
        }
        /// <summary>
        /// 服务接口实现类完全限定名,而且不能有空格，否则无法识别
        /// 形式："完全类名,程序集名,Version=1.0.0,Culture=neutral,PublicKeyToken=null"
        /// </summary>	
        public string ComponentServiceCompeleteName
        {
            get;
            set;
        }
        /// <summary>
        /// 服务接口实现类名
        ///  形式："类名"
        /// </summary>	
        public string ComponentServiceName
        {
            get;
            set;
        }
        /// <summary>
        /// 程序集名称
        /// </summary>	
        public string ComponentAssemblyName
        {
            get;
            set;
        }
        /// <summary>
        /// 程序集文件名称
        /// </summary>	
        public string ComponentAssemblyFileName
        {
            get;
            set;
        }
        /// <summary>
        /// 程序集文件存储路径
        /// 绝对路径
        /// </summary>	
        public string ComponentAssemblyFileStorePath
        {
            get;
            set;
        }
        /// <summary>
        /// 接口名称
        /// </summary>
        public List<string> ComponentInterface
        {
            set;get;

        }     
        /// <summary>
        /// 程序集版本
        /// </summary>		

        public string AssemblyVersion
        {
            get;
            set;
        }
    }
}

File: Hayaa.ServiceFrame/ServiceFrameClient/hayaa.serviceplatform.client_java/hayaa.serviceplatform.client/src/main/java/hayaa/serviceplatform/client/AppComponent.java
package hayaa.serviceplatform.client;

import java.util.List;

class AppComponent {
    private  int AppUserId;
    private int ComponetId;
    private String ComponentServiceCompeleteName;
    private String ComponentServiceName;
    private String ComponentAssemblyName;
    private String ComponentAssemblyFileName;
    private String ComponentAssemblyFileStorePath;
    private List<String> ComponentInterface;
    private String AssemblyVersion;

    public int getAppUserId() {
        return AppUserId;
    }

    public void setAppUserId(int appUserId) {
        AppUserId = appUserId;
    }

    public int getComponetId() {
        return ComponetId;
    }

    public void setComponetId(int componetId) {
        ComponetId = componetId;
    }

    public String getComponentServiceCompeleteName() {
        return ComponentServiceCompeleteName;
    }

    public void setComponentServiceCompeleteName(String componentServiceCompeleteName) {
        ComponentServiceCompeleteName = componentServiceCompeleteName;
    }

    public String getComponentServiceName() {
        return ComponentServiceName;
    }

    public void setComponentServiceName(String componentServiceName) {
        ComponentServiceName = componentServiceName;
    }

    public String getComponentAssemblyName() {
        return ComponentAssemblyName;
    }

    public void setComponentAssemblyName(String componentAssemblyName) {
        ComponentAssemblyName = componentAssemblyName;
    }

    public String getComponentAssemblyFileName() {
        return ComponentAssemblyFileName;
    }

    public void setComponentAssemblyFileName(String componentAssemblyFileName) {
        ComponentAssemblyFileName = componentAssemblyFileName;
    }

    public String getComponentAssemblyFileStorePath() {
        return ComponentAssemblyFileStorePath;
    }

    public void setComponentAssemblyFileStorePath(String componentAssemblyFileStorePath) {
        ComponentAssemblyFileStorePath = componentAssemblyFileStorePath;
    }

    public List<String> getComponentInterface() {
        return ComponentInterface;
    }

    public void setComponentInterface(List<String> componentInterface) {
        ComponentInterface = componentInterface;
    }

    public String getAssemblyVersion() {
        return AssemblyVersion;
    }

    public void setAssemblyVersion(String assemblyVersion) {
        AssemblyVersion = assemblyVersion;
    }


}
